pkg/utils/dataset_test.go

```diff
@@ -0,0 +1,327 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package utils
+
+import (
+	"reflect"
+	"testing"
+
+	datav1alpha1 "github.com/fluid-cloudnative/fluid/api/v1alpha1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+func TestGetDataset(t *testing.T) {
+	mockDatasetName := "fluid-data-set"
+	mockDatasetNamespace := "default"
+	initDataset := &datav1alpha1.Dataset{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      mockDatasetName,
+			Namespace: mockDatasetNamespace,
+		},
+	}
+	s := runtime.NewScheme()
+	s.AddKnownTypes(datav1alpha1.GroupVersion, initDataset)
+
+	fakeClient := fake.NewFakeClientWithScheme(s, initDataset)
+
+	testCases := map[string]struct {
+		name      string
+		namespace string
+		wantName  string
+		notFound  bool
+	}{
+		"get Dataset test case 1": {
+			name:      mockDatasetName,
+			namespace: mockDatasetNamespace,
+			wantName:  mockDatasetName,
+			notFound:  false,
+		},
+		"get Dataset test case 2": {
+			name:      mockDatasetName + "not-exist",
+			namespace: mockDatasetNamespace,
+			wantName:  "",
+			notFound:  true,
+		},
+	}
+
+	for k, item := range testCases {
+		gotDataset, err := GetDataset(fakeClient, item.name, item.namespace)
+		if item.notFound {
+			if err == nil && gotDataset != nil {
+				t.Errorf("%s check failure, want got nil", k)
+			}
+		} else {
+			if gotDataset.Name != item.wantName {
+				t.Errorf("%s check failure,got Dataset name:%s,want name:%s", k, gotDataset.Name, item.wantName)
+			}
+		}
+	}
+
+}
+
+func TestIsSetupDone(t *testing.T) {
+	testCases := map[string]struct {
+		conditions []datav1alpha1.DatasetCondition
+		wantDone   bool
+	}{
+		"test dataset is setup done case 1": {
+			conditions: []datav1alpha1.DatasetCondition{
+				{Type: datav1alpha1.DatasetReady},
+			},
+			wantDone: true,
+		},
+		"test dataset is setup done case 2": {
+			conditions: []datav1alpha1.DatasetCondition{
+				{Type: datav1alpha1.DatasetInitialized},
+			},
+			wantDone: false,
+		},
+		"test dataset is setup done case 3": {
+			conditions: nil,
+			wantDone:   false,
+		},
+	}
+
+	for k, item := range testCases {
+		dataset := mockDatasetWithCondition("dataset-1", "default", item.conditions)
+		gotDone := IsSetupDone(dataset)
+
+		if gotDone != item.wantDone {
+			t.Errorf("%s check failure, want:%t,got:%t", k, item.wantDone, gotDone)
+		}
+
+	}
+
+}
+
+func TestGetAccessModesOfDataset(t *testing.T) {
+
+	testCases := map[string]struct {
+		name            string
+		getName         string
+		namespace       string
+		accessModel     []v1.PersistentVolumeAccessMode
```

Comment:
`accessModes` or `accessMode` instead of `accessModel` which may be a little bit confusing